app/models/host/managed.rb

```diff
@@ -485,6 +485,11 @@ def host_inherited_params_objects
     params
   end
 
+  def host_params_objects
+    # Host parameters should always be first for the uniq order
+    (host_parameters + host_inherited_params_objects).uniq {|param| param.name}
```

Comment:
I think doing it here would be best so we don't change the behaviour of the other (which may be important in the UI or something), just be sure to cover it with a test.